bees/mastodonbee/mastodonbeefactory.go

```diff
@@ -106,14 +106,312 @@ func (factory *mastodonBeeFactory) Events() []bees.EventDescriptor {
 	events := []bees.EventDescriptor{
 		{
 			Namespace:   factory.Name(),
-			Name:        "toot_sent",
-			Description: "A toot has been sent",
+			Name:        "deleted",
+			Description: "is triggered when a toot has been deleted",
 			Options: []bees.PlaceholderDescriptor{
+				{
+					Name:        "id",
+					Description: "The ID of the deleted toot",
+					Type:        "string",
+				},
+			},
+		},
+		{
+			Namespace:   factory.Name(),
+			Name:        "toot_fetched",
+			Description: "is triggered when a toot has been fetched",
+			Options: []bees.PlaceholderDescriptor{
+				{
+					Name:        "id",
+					Description: "The ID of the toot",
+					Type:        "string",
+				},
 				{
 					Name:        "text",
 					Description: "Text of the toot that has been sent",
 					Type:        "string",
 				},
+				{
+					Name:        "user_id",
+					Description: "Mastodon ID if the toots author",
+					Type:        "string",
+				},
+				{
+					Name:        "username",
+					Description: "Mastodon handle of the toots author",
+					Type:        "string",
+				},
+				{
+					Name:        "reblogs",
+					Description: "reblogs count",
+					Type:        "int64",
+				},
+				{
+					Name:        "favourites",
+					Description: "favourites count",
+					Type:        "int64",
+				},
+				{
+					Name:        "url",
+					Description: "The url for the toot",
+					Type:        "string",
+				},
+			},
+		},
+		{
+			Namespace:   factory.Name(),
+			Name:        "follow",
+			Description: "is triggered when someone wants to follow you",
+			Options: []bees.PlaceholderDescriptor{
+				{
+					Name:        "user_id",
+					Description: "Mastodon ID of the user which triggered the follow event",
+					Type:        "string",
+				},
+				{
+					Name:        "username",
+					Description: "Mastdon handle of the user which triggered the follow event",
+					Type:        "string",
+				},
+				{
+					Name:        "following",
+					Description: "Indicates if your're following the user",
+					Type:        "bool",
+				},
+				{
+					Name:        "followed_by",
+					Description: "Indicates if you're followed by the user",
+					Type:        "bool",
+				},
+				{
+					Name:        "followers",
+					Description: "Number of followers for the user which triggered the follow request",
+					Type:        "int64",
+				},
+				{
+					Name:        "follows",
+					Description: "Number of follows for the user which triggered the follow request",
+					Type:        "int64",
+				},
+			},
+		},
+		{
+			Namespace:   factory.Name(),
+			Name:        "followed",
+			Description: "is triggered when you want to follow someone on Mastdon",
```

Comment:
Typo `Mastdon`.

Reply:
Should maybe read: "is triggered when you followed someone on Mastodon".

Reply:
Applied as requested

I wasn't sure on this one. Maybe it should read: "is triggered when you attempt to follow someone on Mastodon"? E.g. This event can also be triggered in case you want to follow someone who approves incoming follow-request by hand (and therefore you're **not yet** following him)